src/data/nat/basic.lean

```diff
@@ -64,6 +64,9 @@ instance : linear_ordered_cancel_add_comm_monoid ℕ :=
 { add_left_cancel := @nat.add_left_cancel,
   ..nat.linear_ordered_semiring }
 
+instance : has_exists_add_of_le ℕ :=
+{ exists_add_of_le := λ a b hab, ⟨b - a, (nat.add_sub_of_le hab).symm⟩ }
```

Comment:
You shouldn't need this instance, a better one would be:
```
instance [canonically_ordered_add_monoid α] : has_exists_add_of_le α :=
{ exists_add_of_le := λ a b hab, (le_iff_exists_add a b).mp hab }
```

Reply:
that depends what happens in your last comment. ``canonically_ordered_add_monoid`` isn't enough to imply that translating intervals is bijective - the 'addition is maximum' monoid on ``nat`` with the usual ordering is a counterexample.

Reply:
It doesn't matter here. All that matters when defining this instance is having what is necessary to prove `exists_add_of_le`.

Reply:
ah, gotcha. about to make some changes then